### PR TITLE
implementing getTopAlbum

### PR DESCRIPTION
**Pull request by coderveda27**
Hi, so I added some javadoc on the user DAO file.

More importantly, I implemented the getTopAlbums in the user DAO file. There currently is an error which I would like to discuss with you all before making the change to resolve it. 

So my thinking process to implement that method was this:  I iterated through the top songs (which are song objects) and used the getAlbum method to get each top song's album. Then I would return a list of all the albums I got. Is this the right thinking? 

The issue is that the return type for getTopAlbums is Song[].  However, in my code I have returned a list of strings (where the strings are albums of the top songs). I did this because albums are defined as strings in the Song constructor. 

So to fix this issue I guess I could change the return type of the method from Song[] to String[] but I believe that would involve editing some other files as well. Or perhaps we could just make an Album entity?

### CODE REVIEW

**Review by lunazay — COMMENTED** [review]
the thought process is correct, the only place where this method is used is in the top albums interactor, so changing the return type to ArrayList<String> should not be a problem it is simply about design, if it is better for your use case to no have a list of Album object and just a list of strings as the output then you should do that. I dont entirely understand why the function would return Song object for album, is there anything in particular blocking you from creating an Album object and making that a new entity?

**Comment by coderveda27** [comment]
thanks luna. I suppose it is a design decision after all. I tried changing the return types and all but I just thought it would be better to add the Album entity. I know we initially did not have that entity since the albums were accessed through the song entity, but I feel having an album entity is more flexible. :) 

I should have fixed any issues from before. As usual, feel free to let me know if anything needs to be fixed.

**Review by ashleychristen — APPROVED** [review]
this looks wonderful thank you for finishing you use case. I appreciate you taking Luna's feedback and fixing the errors you originally had. Thank you for also adding java documentation. I do think that creating an Album entity was better for upholding SOLID principles specifically the Open/Closed principle. Creating this entity also is better for Clean Architecture so we are not making any unnecessary dependencies on the Song entity.

File: src/entity/Song.java [hunk 1]
```diff
@@ -7,22 +7,22 @@ public class Song extends DataObject{
...
     }
 
 
     public List<String> getArtist() {
         return artists;
     }
 
-    public String getAlbum() {
-        return album;
+    public Album[] getAlbums() {
```
Comment (lunazay) [inline src/entity/Song.java:24]: here is Album a list, maybe try rechecking I think it might be a String that youre returning

**Pull request merged**